resources/test_data/sqlite_testrunner_queries.sql

```diff
@@ -419,6 +419,16 @@ SELECT CASE WHEN id + 3.4 < 50 THEN 'Hello' WHEN id < 70 THEN 'World' ELSE 'Ciao
 SELECT CASE id + 10 WHEN 15 THEN a WHEN 26 THEN 'World' ELSE d END AS case_column FROM mixed;
 SELECT a, CASE WHEN a IS NULL THEN 1 ELSE 2 END FROM mixed_null GROUP BY a
 
+-- COALESCE
+SELECT COALESCE(d, '') d_not_null FROM mixed_null;
+SELECT COALESCE(b, -1) AS b_not_null, COALESCE(c, 0.0) c_not_null FROM mixed_null;
+SELECT COALESCE(b, 1-1) b_not_null, COALESCE(c, 3.0/-4.0) c_not_null FROM mixed_null;
+SELECT COALESCE(c, CAST(COALESCE(b, 0) AS FLOAT)) c_not_null FROM mixed_null;
+SELECT COALESCE(b, CAST(c AS INT), 0) b_not_null FROM mixed_null;
```

Comment:
What about `COALESCE(NULL, c, NULL)` on a nullable column? Does it make sense to test it?

Reply:
Can add it, we just get `NULL` then. From the standard:
> d) `COALESCE (V1, V2)` is equivalent to the following `<case specification>`:
>  ```SQL
>  CASE
>    WHEN NOT V1 IS NULL THEN V1 
>    ELSE V2
>  END
>  ```
> e) `COALESCE (V1, V2, ..., Vn)`, for n ≥ 3, is equivalent to the following `<case specification>`:
>  ```SQL
>  CASE
>    WHEN NOT V1 IS NULL THEN V1 
>    ELSE COALESCE (V2, ..., Vn)
>  END
> ```
[SQL:2023-2, p. 251]

Thus, your example is equivalent to:
```SQL
CASE
  WHEN NULL IS NOT NULL THEN NULL  -- okay, not gonna happen
  WHEN c IS NOT NULL THEN c        
  ELSE NULL                        -- so just c
```
or simply `c`. Gonna add that case and also the standard excerpt.

Reply:
Ah, good to know. I wasn't sure how to read SQLServer's description "At least one of the null values must be a typed NULL" (https://learn.microsoft.com/en-us/sql/t-sql/language-elements/coalesce-transact-sql?view=sql-server-ver16). No idea what "typed" means here ...